armada/api/tiller_controller.py

```diff
@@ -13,53 +13,74 @@
 
-from armada.handlers.tiller import Tiller as tillerHandler
+from armada import api
+from armada.handlers.tiller import Tiller
 
-LOG = logging.getLogger(__name__)
-CONF = cfg.CONF
-DOMAIN = "armada"
-
-logging.setup(CONF, DOMAIN)
-
-class Status(object):
+class Status(api.BaseResource):
     def on_get(self, req, resp):
         '''
         get tiller status
         '''
-        message = "Tiller Server is {}"
-        if tillerHandler().tiller_status():
-            resp.data = json.dumps({'message': message.format('Active')})
-            LOG.info('Tiller Server is Active.')
-        else:
-            resp.data = json.dumps({'message': message.format('Not Present')})
-            LOG.info('Tiller Server is Not Present.')
+        try:
+            policy_action = 'chart_installer:query_status'
+            ctx = req.context
+
+            if not self.check_policy(policy_action, ctx):
+                self.access_denied(req, resp, policy_action)
+                return
 
-        resp.content_type = 'application/json'
-        resp.status = HTTP_200
+            message = {
+                'tiller': Tiller().tiller_status()
+            }
 
-class Release(object):
+            if message.get('tiller'):
+                resp.data = json.dumps(message)
+                resp.status = HTTP_200
+            else:
+                resp.data = json.dumps(message)
+                resp.status = HTTP_503
+
+            resp.content_type = 'application/json'
+        except Exception:
+            self.error(req.context, "Unable to find resources")
+            self.return_error(
+                resp,
+                HTTP_503,
+                message="Unable to find resources"
+            )
+
+
+class Release(api.BaseResource):
     def on_get(self, req, resp):
         '''
         get tiller releases
         '''
-        # Get tiller releases
-        handler = tillerHandler()
 
-        releases = {}
-        for release in handler.list_releases():
-            releases[release.name] = release.namespace
+        try:
+
+            policy_action = 'chart_installer:read_releases'
+            ctx = req.context
+
+            if not self.check_policy(policy_action, ctx):
+                self.access_denied(req, resp, policy_action)
+                return
+
+            # Get tiller releases
+            handler = Tiller()
+            releases = {}
+            for release in handler.list_releases():
+                releases[release.name] = release.namespace
 
-        resp.data = json.dumps({'releases': releases})
-        resp.content_type = 'application/json'
-        resp.status = HTTP_200
+            resp.data = json.dumps({'releases': releases})
+            resp.content_type = 'application/json'
+            resp.status = HTTP_200
+        except Exception:
+            self.error(req.context, "Unable to find Armada Releases")
+            self.return_error(
+                resp,
+                HTTP_404,
```

Comment:
A 404 error might be confusing to a client here. If there are no releases available, an HTTP 200 code seems appropriate because the request went through successfully. I would add some feedback to the response if the releases are empty that says something like "no releases could be found". 

As for the exception, an HTTP 500 error seems appropriate, because an error with Tiller services would be a server error. A 404 error implies that there is a client error that led to an endpoint that does not exist.